src/main/host/descriptor/socket/netlink.rs

```diff
@@ -0,0 +1,1215 @@
+            buffer.len()
+        } else {
+            total_copied
+        };
+
+        Ok(RecvmsgReturn {
+            return_val: return_val.try_into().unwrap(),
+            addr: Some(src_addr),
+            msg_flags: 0,
+            control_len: 0,
+        })
+    }
+
+    fn handle_error(&self, bytes: &[u8]) -> Vec<u8> {
+        // If we can't get the pid, set it to zero
+        let nlmsg_seq = match bytes.get(8..12) {
+            Some(x) => u32::from_le_bytes(x.try_into().unwrap()),
+            None => 0,
+        };
+
+        // Generate a dummy error with the same sequence number as the request
+        let msg = {
+            let len = None;
+            let nl_type = Nlmsg::Error;
+            let flags = NlmFFlags::empty();
+            let pid = None;
+            let payload = NlPayload::<Nlmsg, ()>::Empty;
+            Nlmsghdr::new(len, nl_type, flags, Some(nlmsg_seq), pid, payload)
+        };
+
+        let mut buffer = Cursor::new(Vec::new());
+        msg.to_bytes(&mut buffer).unwrap();
+        buffer.into_inner()
+    }
+
+    fn handle_ifaddrmsg(&self, common: &mut NetlinkSocketCommon, bytes: &[u8]) -> Vec<u8> {
+        let Ok(nlmsg) = Nlmsghdr::<Rtm, Ifaddrmsg>::from_bytes(&mut Cursor::new(bytes)) else {
+            log::warn!("Failed to deserialize the message");
+            return self.handle_error(bytes);
+        };
+
+        let Ok(ifaddrmsg) = nlmsg.get_payload() else {
+            log::warn!("Failed to find the payload");
+            return self.handle_error(bytes);
+        };
+
+        // The only supported interface address family is AF_INET
+        if ifaddrmsg.ifa_family != RtAddrFamily::Unspecified
+            && ifaddrmsg.ifa_family != RtAddrFamily::Inet
+        {
+            log::warn!("Unsupported ifa_family (only AF_UNSPEC and AF_INET are supported)");
+            return self.handle_error(bytes);
+        }
+
+        // The rest of the fields are unsupported. We limit only the interest to the zero values
+        if ifaddrmsg.ifa_prefixlen != 0
+            || ifaddrmsg.ifa_flags != IfaFFlags::empty()
+            || ifaddrmsg.ifa_index != 0
+            || ifaddrmsg.ifa_scope != libc::c_uchar::from(RtScope::Universe)
+        {
+            log::warn!(
+                "Unsupported ifa_prefixlen, ifa_flags, ifa_scope, or ifa_index (they have to be 0)"
+            );
+            return self.handle_error(bytes);
+        }
+
+        let mut buffer = Cursor::new(Vec::new());
+        // Send the interface addresses
+        for interface in &common.interfaces {
+            let address = interface.address.octets();
+            let broadcast = if interface.prefix_len == 32 {
+                address
+            } else {
+                Ipv4Addr::from(
+                    (0xffff_ffff_u32) >> u32::from(interface.prefix_len)
+                        | u32::from(interface.address),
+                )
+                .octets()
+            };
```

Comment:
minor nit: I think simpler as just the following, which doesn't need the check for `prefix_len >= 32` to avoid the panic:

```rust
let broadcast = Ipv4Addr::from(
    0xffff_ffff_u32
        .checked_shr(u32::from(interface.prefix_len))
        .unwrap_or(0)
        | u32::from(interface.address),
)
.octets();
```

Reply:
Yeah, good idea